### PR TITLE
[resource_datadog_monitor] Add separate validation for existing monitors

### PR DESCRIPTION
**Pull request by gkharkwal**
Tested this locally with a test tf monitor (see screenshot below). Not quite sure how to test this with Cassettes because we're changing an internal implementation detail without affecting the e2e behavior.

image

### CODE REVIEW

File: datadog/resource_datadog_monitor.go [hunk 1]
```diff
@@ -469,11 +469,22 @@ func resourceDatadogMonitorCustomizeDiff(ctx context.Context, diff *schema.Resou
 	}
 	m, _ := buildMonitorStruct(diff)
 
+	has_id := false
+	id, err := strconv.ParseInt(diff.Id(), 10, 64)
```
Comment (phillip-dd) [inline datadog/resource_datadog_monitor.go:473]: Why not do `has_id = diff.Id() != ""`?

Comment (phillip-dd) [reply datadog/resource_datadog_monitor.go:473]: Oh, `ValidateExistingMonitor` requires an int.

**Review by phillip-dd — COMMENTED** [review]
lgtm

File: datadog/resource_datadog_monitor.go [hunk 2]
```diff
@@ -469,11 +469,22 @@ func resourceDatadogMonitorCustomizeDiff(ctx context.Context, diff *schema.Resou
 	}
 	m, _ := buildMonitorStruct(diff)
 
+	has_id := false
```
Comment (phillip-dd) [inline datadog/resource_datadog_monitor.go:472]: whoops, this should be camelcased  to follow go conventions: `hasID`

**Comment by gkharkwal** [comment]
@jirikuncar I'm seeing errors in datadog/internal/utils/custom_request.go that I don't think I've caused. Can you help with those?

**Comment by jirikuncar** [comment]
@gkharkwal you can rebase on-top of https://github.com/DataDog/terraform-provider-datadog/pull/1409.

**Comment by gkharkwal** [comment]
> @gkharkwal you can rebase on-top of #1409.

Sure thing, done.

**Review by maycmlee — COMMENTED** [review]
Nothing for docs to check.

**Comment by gkharkwal** [comment]
@jirikuncar Mind taking another look when you get a chance?

**Review by therve — APPROVED** [review]

**Pull request merged**